2.0 prep: Remove deprecated flags

jsafrane opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In 2.0 we can safely remove deprecated flags.

**Which issue(s) this PR fixes**:
Related to  #158

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
Action required: the `-connection-timeout`, `-leader-election-type` and `-leader-election-identity` flags, deprecated in v1.2, have been removed. Please update your manifests for the external-attacher.
Action required: the `-dummy` flag has been removed. Please update your manifests for the external-attacher.
```

msau42 commented on cmd/csi-attacher/main.go:
```diff
@@ -57,21 +56,18 @@ const (
+	kubeconfig  = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	resync      = flag.Duration("resync", 10*time.Minute, "Resync interval of the controller.")
+	csiAddress  = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	dummy       = flag.Bool("dummy", false, "Run in dummy mode, i.e. not connecting to CSI driver and marking everything as attached. Expected CSI driver name is \"csi/dummy\".")
```
Was this only for initial development? Can we remove this too?

jsafrane replied:
I removed `-dummy` too, I hope release note generator can handle multi-line comments :-)

jsafrane commented:
I removed `-dummy`. And when at it, I noticed that there was an old unused variable with the old CRD-based informers, so I deleted it (removing lot of files from vendor/)

msau42 commented:
/lgtm
